airflow/providers/snowflake/hooks/snowflake.py

```diff
@@ -242,3 +246,31 @@ def set_autocommit(self, conn, autocommit: Any) -> None:
 
     def get_autocommit(self, conn):
         return getattr(conn, 'autocommit_mode', False)
+
+    def run(self, sql, autocommit=True, parameters=None):
+        """The snowflake python connector library doesn't allow to execute multiple statements from a single string or
+        file and doesn't return any information about the execution. This method will override DBApiHook run method to
+        enable this features using snowflake connector methods"""
+        if isinstance(sql, str):
+            sql = [item[0] for item in split_statements(io.StringIO(sql))]
+
+        with closing(self.get_conn()) as conn:
+            self.set_autocommit(conn, autocommit)
+
+            with closing(conn.cursor(DictCursor)) as cur:
+                for query in sql:
+                    if parameters is not None:
+                        self.log.info(f"{query} with parameters {parameters}")
+                        cur.execute(query, parameters)
+                        for row in cur.fetchall():
+                            self.log.info(f'Statement Execution Info - {row}')
+                    else:
+                        self.log.info(query)
+                        cur.execute(query)
+                        for row in cur.fetchall():
```

Comment:
```suggestion
                        for row in cur:
```

More memory efficient.